hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/spark/sql/hudi/command/CreateHoodieTableCommand.scala

```diff
@@ -88,6 +92,20 @@ case class CreateHoodieTableCommand(table: CatalogTable, ignoreIfExists: Boolean
 
 object CreateHoodieTableCommand {
 
+  def validateTableSchema(userDefinedSchema: StructType, hoodieTableSchema: StructType): Boolean = {
+    if (userDefinedSchema.fields.length != 0 &&
```

Comment:
you  can not create a table two times, you can alter existing table with `alter table` cmd, or use `create table if not exists` to skip the creation.

Reply:
Thanks for the explanation, what should the second `create table` cmd behavior, should we skip the creation or throws error or change the table schema as the new one.

Reply:
@huangxiaopingRD do you mean `ALTER TABLE table_xxx` should delete the table in Spark SQL?

Reply:
`CREATE TABLE` should only be executed if the table does not exist.

Reply:
@yihua @danny0405 
Let me re-describe this problem:
**Scenario:**
1. The user creates a hudi table `table_xxx`, which has three columns ` (id string, name string, list string)`
2. The user finds that the type of the `list` column type should be `map<string, string>` instead of `string`, so he executes the drop table operation, intending to delete the table and recreate it, but the table executed is `drop table table_xxx`, not `drop table table_xxx purge`.  **Without `purge` operation, `.hoodie` directory will not be deleted.**
3. The user re-creates the hudi table `table_xxx`, and the three columns written by the user are `(id string, name string, list map<string, string>)`, but the columns synchronized to HMS are still `(id string, name string, list string)`, because the schema synchronized to HMS is the schema in the `.hoodie` directory, which is the old schema, that is `(id string, name string, list string)`
4. The user writes data to the table` table_xxx`. Because the written data is inconsistent with the schema, SQL parsing fails, and the exception is `org.apache.spark.sql.AnalysisException: Cannot write incompatible data to table 'table_xxx'`

**PR purpose:**
When the schema in the `create table` statement is inconsistent with the schema of the existing `.hoodie` directory, it is forbidden to create a new table and an exception is thrown.

Reply:
Reasonable.